src/guides/legend.jl

```diff
@@ -88,6 +87,11 @@ function _Legend_(fg::FigureGrid)
     labels = Vector{String}[]
     elements_list = Vector{Vector{LegendElement}}[]
 
+    # search in attributes if a color is given for e.g. lines/scatter
+    legend_colors = unique([entry.attributes[:color] for entry in entries(grid) if haskey(entry.attributes, :color)])
```

Comment:
I think this is not completely correct. What if one scatter layer has a color attribute, and another scatter layer has no color attribute, and is defaulting to the theme?

I think one should instead check that, for a given `plottype`, all the entries which use the `marker` scale (or whichever relevant scale is used in that legend group) also have a color attribute, and all the values agree. This may be not too hard to check, it should be all the entries where at least one of the `label_attrs`  is in the keys of `entry.primary`.

I'm also not sure if it should only be done for `color`, or there are other attributes in `visual` that should be considered.